doc/developer-guides/coding_guidelines.rst

```diff
@@ -1121,6 +1428,312 @@ Compliant example::
+============================================
+
+Compliant example::
+
+    uint32_t showcase_u32;
+    uint64_t showcase_u64 = 64UL;
+    
+    showcase_u32 = (uint32_t)showcase_u64;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase_u32;
+       uint64_t showcase_u64 = 64UL;
+       
+       showcase_u32 = showcase_u64;
+
+
+TY-13: A complex integer expression shall not be cast to a wider type
+=====================================================================
+
+Compliant example::
+
+    uint32_t showcase_u32_1 = 10U;
+    uint32_t showcase_u32_2 = 10U;
+    uint64_t showcase_u64;
+    
+    showcase_u64 = (uint64_t)showcase_u32_1 + (uint64_t)showcase_u32_2;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase_u32_1 = 10U;
+       uint32_t showcase_u32_2 = 10U;
+       uint64_t showcase_u64;
+       
+       showcase_u64 = (uint64_t)(showcase_u32_1 + showcase_u32_2);
+
+
+TY-14: A complex signed integer expression shall not be cast to unsigned type
+=============================================================================
+
+Compliant example::
+
+    int32_t showcase_signed_1 = 10;
+    int32_t showcase_signed_2 = 5;
+    uint32_t showcase_unsigned;
+    
+    showcase_unsigned = (uint32_t)showcase_signed_1 - (uint32_t)showcase_signed_2;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       int32_t showcase_signed_1 = 10;
+       int32_t showcase_signed_2 = 5;
+       uint32_t showcase_unsigned;
+       
+       showcase_unsigned = (uint32_t)(showcase_signed_1 - showcase_signed_2);
+
+
+TY-15: A complex unsigned integer expression shall not be cast to signed type
+=============================================================================
+
+Compliant example::
+
+    uint32_t showcase_unsigned_1 = 10U;
+    uint32_t showcase_unsigned_2 = 5U;
+    int32_t showcase_signed;
+    
+    showcase_signed = (int32_t)showcase_unsigned_1 - (int32_t)showcase_unsigned_2;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase_unsigned_1 = 10U;
+       uint32_t showcase_unsigned_2 = 5U;
+       int32_t showcase_signed;
+       
+       showcase_signed = (int32_t)(showcase_unsigned_1 - showcase_unsigned_2);
+
+
+TY-16: A complex integer expression shall not be cast to types other than integer
+=================================================================================
+
+Compliant example::
+
+    /* 0x61 is 'a' in ASCII Table */
+    uint32_t showcase_u32;
+    char showcase_char;
+    
+    showcase_u32 = 0x61U + 1U;
+    showcase_char = (char)showcase_u32;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* 0x61 is 'a' in ASCII Table */
+       uint32_t showcase_u32;
+       char showcase_char;
+       
+       showcase_u32 = 0x61U;
+       showcase_char = (char)(showcase_u32 + 1U);
+
+
+TY-17: Integer shall not be used when a plain char is expected
```

Comment:
Is the word ``plain`` needed here?

Reply:
I will replace `plain char` with `character` in next version.